src/test/java/pl/nogacz/snake/RandomBlocksTest.java

```diff
@@ -0,0 +1,163 @@
+package pl.nogacz.snake;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+import pl.nogacz.snake.board.Board;
+import pl.nogacz.snake.board.Coordinates;
+import pl.nogacz.snake.application.Design;
+
+
+public class RandomBlocksTest {
+    
+    @Test
+    void testSpawnRottenApple1(){
+
+        Design design = mock(Design.class);
+        Board b = new Board(design);
+
+        b.lastSpawnTimesOfApples[0] = System.currentTimeMillis();
+        b.newRandomSpawnTimesOfApples[0] = 0;
+        b.spawnRottenApple();
+        
+        b.lastSpawnTimesOfApples[0] = System.currentTimeMillis();
+        b.newRandomSpawnTimesOfApples[0] = 0;
+        b.spawnRottenApple();
+        
+        b.lastSpawnTimesOfApples[0] = System.currentTimeMillis();
+        b.newRandomSpawnTimesOfApples[0] = 0;
+        b.spawnRottenApple();
+        
+        assertEquals(3, b.rottenApples.size());
+        
+    }
+
+    @Test
+    void testSpawnRottenApple2(){
+
+        Design design = mock(Design.class);
+        Board b = new Board(design);
+
+        b.lastSpawnTimesOfApples[1] = System.currentTimeMillis();
```

Comment:
Bu API çok karışık.

spawnTottenApple() diyorsun ama üstteki testte başka bir obje spawn olurken burada başka bir obje spawn oluyor.

Bence bu APIyi daha temiz bir hale getirmelisin.

b.spawnAbc() => bu Abc objesi spawn etsin
b.spawnDef() => bu Def objesi
b.spawnGhi() => bu da Ghi objesi

Bu spawn ettigin objelere daha manali isimler bul. Apple2, Apple3 kötü isimler.

spawnAbc metodunun içerisinde lastSpawnTimesOfApples ve newRandomSpawnTimesOfApples vs. ne gerekliyse yap. Bu yaptığın atamaların spawn metodunun dışında yapılması bana mantıklı gelmedi.

Reply:
İsimleri revize ediyorum hocam. Burada yapmamın sebebi çağırdığım metodun if conditionlarini saglayarak test edilebilir hale getirilmesiydi.

Reply:
"Bir de gerekirse 10 çeşit yeni özel itemı kolaylıkla ekleyebileceğimiz bir şekilde kod esnek tasarlanmış olmalı." Hocam bu bahsettiğiniz durumu rotten appleları(bloklar) indexleyerek sağlayabildim. Bu nedenle tek bir metot üzerinden verilen parametreye göre rahatça yeni blok entegre edebiliyorum, yeni bir metot yazmadan. Bir metot üzerinden farklı objelerin spawnlanmasının nedeni bu hocam.